official/utils/misc/distribution_utils.py

```diff
@@ -246,3 +246,21 @@ def configure_cluster(worker_hosts=None, task_index=-1):
   else:
     num_workers = 1
   return num_workers
+
+
+class MaybeDistributionScope(object):
```

Comment:
Why not re-use what you added for resnet? 
https://github.com/tensorflow/models/blob/master/official/resnet/keras/keras_imagenet_main.py#L127

Reply:
Good idea.